test/e2e-tests.sh

```diff
@@ -91,6 +103,7 @@ function run_e2e() {
   fi
 }
 
+add_spire "$PIPELINE_FEATURE_GATE"
```

Comment:
Shouldn't this line come after `set_feature_gate "$PIPELINE_FEATURE_GATE"`?

Reply:
The order does not matter as we are not depending on whether spire feature is enabled or not to install SPIRE, which is what add_spire() does.

Reply:
$PIPELINE_FEATURE_GATE is an env variable which is passed in into the e2e-tests.sh script (or rather set before e2e-tests.sh). This can be alpha or stable. 
We set the feature flags based on this env variable.
Both add_spire and set_feature_gate depend on this env variable. set_feature_gate sets the feature_flags config based on this env variable.